agreement/message_test.go

```diff
@@ -123,6 +124,7 @@ func TestMessageBackwardCompatibility(t *testing.T) {
 	e1 := protocol.Encode(&m1)
 	e2 := protocol.EncodeReflect(&m2)
 	require.Equal(t, e1, e2)
+	require.NotContainsf(t, string(e1), "MessageHandle", "encoded message still contains MessageHandle field")
```

Comment:
is it worth writing another test to confirm that it can optionally be included, that way when it's removed the test will break?